controllers/license/konglicense_controller.go

```diff
@@ -36,6 +36,67 @@ import (
 // KongV1Alpha1 KongLicense - Reconciler
 // -----------------------------------------------------------------------------
 
+// NewKongV1Alpha1KongLicenseReconciler creates a new KongV1Alpha1KongLicenseReconciler,
+// that provides functionality for validating the license. The method `GetLicense` returns
+// the license and its validity.
+func NewKongV1Alpha1KongLicenseReconciler(
```

Comment:
How about adding a `WithValidator` method to the reconciler?

Reply:
I changed the approach, got rid of two types for reconciller, changed this parameter to `mo.Optional` and better documented it